### ISSUE TITLE
DRRIP Incorrectly Updates RRPVs

### ISSUE THREAD

**Issue by Quangmire** [ChampSim/ChampSim#505]
**Describe the bug**
DRRIP currently updates RRPVs as follows:
```cpp
auto victim = std::max_element(begin, end);
for (auto it = begin; it != end; ++it)
  *it += ::maxRRPV - *victim; // Bug
```
After `it == victim`, all subsequent RRPV updates are 0.

**Expected behavior**
```cpp
auto victim = std::max_element(begin, end);
auto rrpv_update = ::maxRRPV - *victim;
for (auto it = begin; it != end; ++it)
  *it += rrpv_update;

**Comment by ngober** [comment]
Oh, that's a sneaky bug. Good catch.

**Comment by ngober** [comment]
Resolved by #506.
